### PR TITLE
Utilties+LibCore: Port remaining utilities to LibMain

### LINKED ISSUE #11609: List of Userland Utilities that need to be ported to LibMain
### Required and Recommended targets
- [x] abench
- [x] adjtime (#11961)
- [x] allocate
- [x] aplay
- [x] arp
- [x] asctl
- [x] base64
- [x] basename
- [x] beep (#11752)
- [x] blockdev
- [x] bt
- [x] cal
- [x] cat
- [x] checksum
- [x] chgrp
- [x] chmod
- [x] chown
- [x] chres
- [x] cksum
- [x] clear
- [x] comm
- [x] config
- [x] copy
- [x] cp
- [x] cpp-lexer
- [x] cpp-parser
- [x] cpp-preprocessor
- [x] cut
- [x] date
- [x] dd
- [x] ddate
- [x] df
- [x] diff
- [x] dirname
- [x] disasm (#11880)
- [x] disk_benchmark (#13187)
- [x] dmesg
- [x] du
- [x] echo
- [x] env
- [x] errno
- [x] expr
- [x] false (#11880)
- [x] fdtdump
- [x] fgrep
- [x] file
- [x] find
- [x] flock
- [x] fortune
- [x] functrace
- [x] gml-format (#11880)
- [x] grep (#11880)
- [x] gron
- [x] groupadd (#11579)
- [x] groupdel (#11920)
- [x] groups
- [x] gunzip (#11880)
- [x] gzip (#11721)
- [x] head (#11880)
- [x] hexdump (#11880)
- [x] host (#11880)
- [x] hostname (#11880)
- [x] id
- [x] ifconfig (#11853)
- [x] ini
- [x] jp
- [x] js
- [ ] kcov-example
- [x] keymap
- [x] kill (#12451)
- [x] killall (#12646)
- [x] less
- [x] ln (#11664)
- [x] logout
- [x] ls
- [x] lsirq (#12221)
- [x] lsof (#12221)
- [x] lspci
- [x] lsusb
- [x] man
- [ ] markdown-check
- [x] matroska (#13172)
- [x] md (#12091)
- [x] mkdir
- [x] mkfifo (#11896)
- [x] mknod (#11896)
- [x] mktemp (#11896)
- [x] mount (#12221)
- [x] mv (#12091)
- [x] nc (#11921)
- [x] netstat
- [x] nl (#12221)
- [x] notify (#12221)
- [x] nproc
- [x] ntpquery (#12221)
- [x] open  (#12221)
- [x] pape  (#12221)
- [x] passwd
- [x] paste (#12646)
- [ ] pathchk
- [x] pgrep (#12367)
- [x] pidof (#12367)
- [x] ping (#12159)
- [x] pls
- [x] pmap
- [x] printf (#13172)
- [x] pro
- [x] profile (#12469)
- [x] ps
- [x] purge (#13172)
- [x] pwd
- [x] readelf (#13172)
- [ ] readlink
- [x] realpath (#12377)
- [x] reboot (#12865)
- [x] rev (#12088)
- [x] rm (#12124)
- [x] rmdir (#12124)
- [ ] run-tests
- [x] seq (#11926)
- [x] shot (#12034)
- [x] shuf (#12088)
- [x] shutdown (#12034)
- [x] sleep (#12034)
- [x] sort (#12088)
- [x] sql (#12034)
- [x] stat
- [x] strace
- [x] stty (#12034)
- [x] su
- [x] sync (#12034)
- [x] syscall (#12034)
- [x] sysctl (#12034)
- [x] tac (#11736)
- [x] tail (#12091)
- [ ] tar
- [x] tee (#13246)
- [ ] telws
- [ ] test-bindtodevice
- [x] test-crypto (removed in #11482)
- [ ] test-fuzz
- [x] test-imap (#11292)
- [ ] test-pthread
- [ ] test-unveil
- [ ] test_env
- [x] top
- [x] touch
- [ ] tr
- [x] traceroute (#13172)
- [x] tree (#12123)
- [x] true (#13172)
- [x] truncate
- [ ] tt
- [ ] tty
- [x] umount (#12012)
- [x] uname (#11607)
- [x] uniq (#11607)
- [x] unzip (#11586)
- [x] uptime (#11559)
- [x] useradd (#11920)
- [x] userdel
- [x] usermod
- [x] utmpupdate
- [x] w
- [x] wasm (#11736)
- [x] watch
- [x] wc (#11607)
- [x] which (#11607)
- [x] whoami
- [x] wsctl (#11607)
- [x] xargs (#11607)
- [x] yes (#11607)
- [x] zip (#11586) 

### Special Targets (SPECIAL_TARGETS in cmake file)
- [ ] test

### PR DESCRIPTION
**Pull request by kennethmyhra**
This ports the remaining utilities to LibMain and includes some other minor enhancements and refactorings to SerenityOS patterns.

Closes #11609

### CODE REVIEW

File: Userland/Utilities/test-bindtodevice.cpp [hunk 1]
```diff
@@ -1,43 +1,43 @@
...
-    int fd = socket(AF_INET, SOCK_DGRAM, IPPROTO_UDP);
-    if (fd < 0) {
-        perror("socket");
-        return;
-    }
-
+    auto fd = TRY(Core::System::socket(AF_INET, SOCK_DGRAM, IPPROTO_UDP));
     test_fn(fd);
 
     // be a responsible boi
```
Comment (kleinesfilmroellchen) [inline Userland/Utilities/test-bindtodevice.cpp:27]: 🦬😘

**Review by kleinesfilmroellchen — APPROVED** [review]
Looking great!

**Comment by mjz19910** [comment]
The test utility was missed (test.cpp does not use LibMain yet)

**Comment by kennethmyhra** [comment]
@mjz19910 I will take a look at it

**Pull request merged**